src/DataCollection.R

```diff
@@ -1,45 +1,60 @@
+# Loads data from betydb.org if possible; if not possible,
+# loads data from CSV file in data/ directory.
+
 library(traits)
-library(dplyr)
 library(readr)
-library(DBI)
-library(RPostgreSQL)
 
 options(#betydb_key = readLines('~/.betykey', warn = FALSE),
   betydb_url = "https://betydb.org",
   betydb_api_version = 'beta')
 
-#Gather species data:
-if (!file.exists('data/species.rds')) {
-  species <- betydb_query(table = 'species', limit = 'none') %>% 
-  mutate(specie_id = id)
-  saveRDS(species, 'data/species.rds')
-} else {
-  species <- readRDS('data/species.rds')
+setwd('~/TeamTwo')
+
+betydbQuerySlick <- function(tableName,idName) {
+  # Given tableName, returns data from <tableName> at betydb.org,
+  # and both loads data into environment and saves file to RDS.
+  # If RDS already exists, loads from RDS instead of querying
+  # database. Renames key column 'id' with idName.
+  if (!file.exists(paste0('data/',tableName,'.rds'))) {
+    tableData <- betydb_query(table = tableName, limit = 'none')
+    names(tableData)[names(tableData)=='id'] <- idName
+    saveRDS(tableData, paste0('data/',tableName,'.rds'))
+  } else {
+    tableData <- readRDS(paste0('data/',tableName,'.rds'))
+  }
+  return(tableData)
 }
 
-#Gather sites data:
-if (!file.exists('data/sites.rds')) {
-  sites <- betydb_query(table = 'sites', limit = 'none') %>% 
-  mutate(site_id = id)
-  saveRDS(sites,'data/sites.rds')
-} else {
-  sites <- readRDS('data/sites.rds')
+csvLoadSlick <- function(tableName) {
+  # Given tableName, loads <tableName>.csv from data/, loads
+  # into memory, and saves to RDS. If RDS already exists, loads
+  # from RDS instead of loading from CSV. Sets all column names
+  # to lower case.
+  if (!file.exists(paste0('data/',tableName,'.rds'))) {
+    tableData <- read_csv(paste0('data/',tableName,'.csv'))
+    colnames(tableData) <- tolower(colnames(tableData))
+    saveRDS(tableData,paste0('data/',tableName,'.rds'))
+  } else {
+    tableData <- readRDS(paste0('data/',tableName,'.rds'))
+  }
+  return(tableData)
 }
 
+#Gather species data:
+species <- betydbQuerySlick(tableName = 'species', idName = 'specie_id')
```

Comment:
The ids should use the singular form to match primary keys in traits table. Like `variable_id`